### ISSUE TITLE
[a11y]: tab overflow button has aria errors

### ISSUE THREAD

**Issue by kglickman** [carbon-design-system/carbon#13786]
### Package

@carbon/react

### Package version

1.23

### React version

16.4

### Automated testing tool and ruleset

IBM accessibility checker

### Description

Accessibility checker is reporting error on overflow button of the tabs
Element "button" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true' 

![image](https://github.com/carbon-design-system/carbon/assets/30597411/db846a62-35f0-433e-ba55-1987dbcebf02)

### Reproduction/example

see storybook

### Steps to reproduce

repro in story book with narrow window so that the tabs overflow.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@kglickman Yes thanks for reporting, I think the buttons should have `tabIndex="-1"` as they did in v10. We had to rewrite this overflow behavior as part of v11 and must have missed this detail.
